From 4f65117d8ade16dee22f93e950a1a8521fbf2036 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Fri, 15 Mar 2019 10:53:35 +0800
Subject: [PATCH 2/4] MLK-21152-1 arm64: dts: imx8qxp: correct isi irq type
 setting

commit 166231393fc5163375e9eb36da38955142a072b3 from
https://source.codeaurora.org/external/imx/linux-imx.git

GIC irqchip driver introduces the warning when IRQ type is
set to IRQ_TYPE_NONE while it is NOT belonging to GIC_IRQ_TYPE_PARTITION,
the ISI irqs' setting trigger below WARNING during kernel boot
up, set ISI irqs type to IRQ_TYPE_LEVEL_HIGH to avoid such
warning during kernel bootup, the GIC driver ONLY supports
IRQ_TYPE_LEVEL_HIGH and IRQ_TYPE_EDGE_RISING for SPIs actually.

drivers/irqchip/irq-gic-v3.c:
WARN_ON(*type == IRQ_TYPE_NONE &&
    fwspec->param[0] != GIC_IRQ_TYPE_PARTITION);

[    0.543404] WARNING: CPU: 1 PID: 1 at drivers/irqchip/irq-gic-v3.c:975
		 gic_irq_domain_translate+0xa8/0x110
[    0.552725] Modules linked in:
[    0.555770] CPU: 1 PID: 1 Comm: swapper/0 Not tainted 4.19.26-03309-g4fed199-dirty #230
[    0.563756] Hardware name: Freescale i.MX8QXP MEK (DT)
[    0.568883] pstate: 00000005 (nzcv daif -PAN -UAO)
[    0.573663] pc : gic_irq_domain_translate+0xa8/0x110
[    0.578619] lr : irq_create_fwspec_mapping+0x64/0x33c
[    0.583650] sp : ffff00000805b890
[    0.586950] x29: ffff00000805b890 x28: ffff80083be9d440
[    0.592249] x27: 0000000000000000 x26: 0000000000000001
[    0.597548] x25: ffff80083be9d440 x24: ffff0000094b8000
[    0.602848] x23: 0000000000000000 x22: ffff00000805b908
[    0.608147] x21: ffff80083ff93188 x20: ffff0000094b8000
[    0.613446] x19: ffff80083c81c000 x18: 000000000000000e
[    0.618746] x17: 0000000000000001 x16: 0000000000000019
[    0.624045] x15: 0000000000000033 x14: 000000000000004c
[    0.629344] x13: 00000000000002fd x12: 0000000000000000
[    0.634644] x11: 0000000000000020 x10: 0101010101010101
[    0.639943] x9 : 0000000000000001 x8 : 7f7f7f7f7f7f7f7f
[    0.645242] x7 : 0000000000000000 x6 : 0000000000000051
[    0.650542] x5 : ffff000008485f14 x4 : ffff000008f9a388
[    0.655841] x3 : ffff00000805b8cc x2 : 0000000000000000
[    0.661141] x1 : 00000000a110c8ee x0 : 0000000000000000
[    0.666442] Call trace:
[    0.668879]  gic_irq_domain_translate+0xa8/0x110
[    0.673481]  irq_create_of_mapping+0x80/0xa8
[    0.677741]  of_irq_get+0x70/0xe4
[    0.681039]  of_irq_to_resource+0x38/0xfc
[    0.685036]  of_irq_to_resource_table+0x4c/0x68
[    0.689555]  of_device_alloc+0x120/0x1b8
[    0.693461]  of_platform_device_create_pdata+0x7c/0x11c
[    0.698674]  of_platform_bus_create+0x140/0x388
[    0.703191]  of_platform_bus_create+0x1a0/0x388
[    0.707711]  of_platform_populate+0x50/0xd4
[    0.711882]  of_platform_default_populate_init+0xbc/0xd0
[    0.717181]  do_one_initcall+0x58/0x168
[    0.721004]  kernel_init_freeable+0x1b4/0x274
[    0.725347]  kernel_init+0x10/0x104
[    0.728821]  ret_from_fork+0x10/0x18
[    0.732387] ---[ end trace 7132294cf9624401 ]---

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
index fa5a9a037249..c9afad4e6ee0 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
@@ -2118,7 +2118,7 @@
 		isi_0: isi@58100000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58100000 0x0 0x10000>;
-			interrupts = <0 297 0>;
+			interrupts = <0 297 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <2 0 2>;  /* <Input MIPI_VCx Output>
 									Input:  0-DC0, 1-DC1, 2-MIPI CSI0, 3-MIPI CSI1, 4-HDMI, 5-MEM
 									VCx:    0-VC0, 1-VC1, 2-VC2, 3-VC3, MIPI CSI only
@@ -2134,7 +2134,7 @@
 		isi_1: isi@58110000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58110000 0x0 0x10000>;
-			interrupts = <0 298 0>;
+			interrupts = <0 298 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <2 1 2>;
 			clocks = <&clk IMX8QXP_IMG_PDMA_1_CLK>;
 			clock-names = "per";
@@ -2147,7 +2147,7 @@
 		isi_2: isi@58120000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58120000 0x0 0x10000>;
-			interrupts = <0 299 0>;
+			interrupts = <0 299 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <2 2 2>;
 			clocks = <&clk IMX8QXP_IMG_PDMA_2_CLK>;
 			clock-names = "per";
@@ -2160,7 +2160,7 @@
 		isi_3: isi@58130000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58130000 0x0 0x10000>;
-			interrupts = <0 300 0>;
+			interrupts = <0 300 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <2 3 2>;
 			clocks = <&clk IMX8QXP_IMG_PDMA_3_CLK>;
 			clock-names = "per";
@@ -2173,7 +2173,7 @@
 		isi_4: isi@58140000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58140000 0x0 0x10000>;
-			interrupts = <0 301 0>;
+			interrupts = <0 301 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <3 0 2>;
 			clocks = <&clk IMX8QXP_IMG_PDMA_4_CLK>;
 			clock-names = "per";
@@ -2186,7 +2186,7 @@
 		isi_5: isi@58150000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58150000 0x0 0x10000>;
-			interrupts = <0 302 0>;
+			interrupts = <0 302 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <3 1 2>;
 			clocks = <&clk IMX8QXP_IMG_PDMA_5_CLK>;
 			clock-names = "per";
@@ -2199,7 +2199,7 @@
 		isi_6: isi@58160000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58160000 0x0 0x10000>;
-			interrupts = <0 303 0>;
+			interrupts = <0 303 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <3 2 2>;
 			clocks = <&clk IMX8QXP_IMG_PDMA_6_CLK>;
 			clock-names = "per";
@@ -2212,7 +2212,7 @@
 		isi_7: isi@58170000 {
 			compatible = "fsl,imx8-isi";
 			reg = <0x0 0x58170000 0x0 0x10000>;
-			interrupts = <0 304 0>;
+			interrupts = <0 304 IRQ_TYPE_LEVEL_HIGH>;
 			interface = <3 3 2>;
 			clocks = <&clk IMX8QXP_IMG_PDMA_7_CLK>;
 			clock-names = "per";
-- 
2.17.1

